lib/active_merchant/billing/gateways/jetpay.rb

```diff
@@ -200,13 +200,18 @@ def scrub(transcript)
 
       private
 
-      def build_xml_request(transaction_type, transaction_id = nil, &block)
+      def build_xml_request(transaction_type, options = {}, transaction_id = nil, &block)
         xml = Builder::XmlMarkup.new
         xml.tag! 'JetPay' do
           # The basic values needed for any request
           xml.tag! 'TerminalID', @options[:login]
           xml.tag! 'TransactionType', transaction_type
           xml.tag! 'TransactionID', transaction_id.nil? ? generate_unique_id.slice(0, 18) : transaction_id
+          if options && options[:origin]
+            xml.tag! 'Origin', options[:origin]
+          else
```

Comment:
@rwdaigle I'm curious about INTERNET being the default.  Do you think we need a default for this field?  I'm afraid this might break folks who don't want the field specified.